bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -437,12 +437,20 @@ func (r *KubeadmConfigReconciler) handleClusterNotInitialized(ctx context.Contex
 	}
 
 	certificates := secret.NewCertificatesForInitialControlPlane(scope.Config.Spec.ClusterConfiguration)
-	err = certificates.LookupOrGenerate(
-		ctx,
-		r.Client,
-		util.ObjectKey(scope.Cluster),
-		*metav1.NewControllerRef(scope.Config, bootstrapv1.GroupVersion.WithKind("KubeadmConfig")),
-	)
+
+	// If the Cluster does not have a ControlPlaneReference look up and generate the certificates.
```

Comment:
For keeping track of the discussion
```suggestion
        // If the Cluster does not have a ControlPlane reference look up and generate the certificates,
        // otherwise rely on certificates generated by the ControlPlane controller.
	// Note: A cluster doe not have ControlPlane reference when using standalone CP machines.
```